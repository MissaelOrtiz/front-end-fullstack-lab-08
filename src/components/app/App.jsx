import React from 'react';
import GamesList from '../games/GameList';

export default function App() {
  return <GamesList />;
}
