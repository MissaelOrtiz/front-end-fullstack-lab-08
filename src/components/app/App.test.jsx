/* eslint-disable max-len */
import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import App from './App';

const server = setupServer(
  rest.get(
    'https://ac-vill.herokuapp.com/villagers/',
    (req, res, ctx) => {
      return res(
        ctx.json([{ id: 1, title: 'PF2e', medium: 'TTRPG', genre: 'adventure' }])
      );
    }
  )
);

describe('App test', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('renders a list of games', async () => {
    const { container } = render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );

    screen.getByText('Now Loading');

    const ul = await screen.findByRole('list', { name: 'games' });
    expect(ul).not.toBeEmptyDOMElement();
    expect(container).toMatchSnapshot();
  });
});
