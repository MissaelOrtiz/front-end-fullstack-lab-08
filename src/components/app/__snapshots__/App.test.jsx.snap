// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test renders a list of games 1`] = `
<div>
  <header>
    <nav>
      <a
        href="/"
      >
         Home 
      </a>
      <a
        href="/create"
      >
         Create 
      </a>
    </nav>
  </header>
  <ul
    aria-label="games"
    role="list"
  >
    <li>
      <button>
        <h2>
          PF2e
        </h2>
      </button>
    </li>
  </ul>
</div>
`;
